pkg/kubectl/resource/builder.go

```diff
@@ -707,7 +707,6 @@ func (b *Builder) resourceTupleMappings() (map[string]*meta.RESTMapping, error)
 			return nil, err
 		}
 
-		mappings[mapping.Resource] = mapping
```

Comment:
Could you explain why did we have this line before? Can `mapping.Resource` and `r.Resource` be different?

Seems to be fine though since the only place that [`resourceTupleMappings()` is called](https://github.com/deads2k/kubernetes/blob/1cb797e355f75a14c4fdd69d396331d845353b76/pkg/kubectl/resource/builder.go#L825) uses `resourceTuples` as key

Reply:
> Could you explain why did we have this line before? Can mapping.Resource and r.Resource be different?
> 
> Seems to be fine though since the only place that resourceTupleMappings() is called uses resourceTuples as key

It was an optimization to avoid building extra clients from back when we ran a heavy-weight discovery to create a client.